src/html_tag.cpp

```diff
@@ -2751,10 +2751,15 @@ litehtml::element::ptr litehtml::html_tag::get_child( int idx ) const
 
 size_t litehtml::html_tag::get_index() const
 {
+	return get_index( this );
+}
+
+size_t litehtml::html_tag::get_index( const litehtml::element * element ) const
+{

```

Comment:
The behaviour is modeled after what I read here: https://api.jquery.com/index/
 * Using the `this->m_Parent` is desired, because you want to check where the element that you passed as an argument, is located among ***your*** siblings
* If `element->m_Parent` is different than `this->m_Parent` it will not be found, so -1 will be returned.
* for the 3th question, see the documentation for the jquery index()